ci/docker-push.yml

```diff
@@ -0,0 +1,71 @@
+name: Docker
+
+on:
+  push:
+    # Publish `master` as Docker `latest` image.
+    branches:
+      - master
+
+    # Publish `v1.2.3` tags as releases.
+    tags:
+      - v*
+
+  # Run tests for any PRs.
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+env:
+  # TODO: Change variable to your repository's name.
+  IMAGE_NAME: image
+
+jobs:
+  # Run tests.
+  # See also https://docs.docker.com/docker-hub/builds/automated-testing/
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Build
+        run: docker-compose -f docker-compose.test.yml build
+
+      - name: Test
+        run: docker-compose -f docker-compose.test.yml run sut
+
+  # Push image to GitHub Package Registry.
+  # See also https://docs.docker.com/docker-hub/builds/
+  push:
+    # Ensure test job passes before pushing image.
+    needs: test
+
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Build
+        run: docker build . --file Dockerfile --tag image
+
+      - name: Login
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u ${{ github.actor }} --password-stdin
+
+      - name: Push
+        run: |
+          IMAGE_ID=docker.pkg.github.com/${{ github.repository }}/$IMAGE_NAME
+
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
```

Comment:
I was hoping there would be a more straight forward way to access the "short revision" you'd get out of something like `git rev-parse --short HEAD`. Maybe supporting `${{ github.short_ref }}` could be a handy feature to upstream.

Reply:
@josh have you considered `VERSION="$(git rev-parse --short HEAD)"`?

Reply:
oops nm i misunderstood